### PR TITLE
Store creation MVP - account creation: Yosemite layer changes

### PR DESCRIPTION
**Pull request by jaclync**
Yosemite layer for #7891
⚠️ Please review the base PR https://github.com/woocommerce/woocommerce-ios/pull/7897 first

### Description

This PR includes the Yosemite layer changes for store creation MVP with one action `AccountCreationAction.createAccount` and implementation in `AccountCreationStore`.

The WPCOM client ID and secret are passed during initialization of `AccountCreationStore`. `createAccount` internally calls remote `loadUsernameSuggestions` given the user email to generate a valid username, and then creates an account given the user email, auto-generated username, and the password.

### Testing instructions

Just CI for now, the new functions aren't used in the app yet.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/AccountCreationStore.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        switch action {
+        case .createAccount(let email, let password, let completion):
+            createAccount(email: email, password: password, completion: completion)
+        }
+    }
+}
+
+private extension AccountCreationStore {
+    func createAccount(email: String, password: String, completion: @escaping (Result<CreateAccountResult, CreateAccountError>) -> Void) {
+        Task { @MainActor in
```
Comment (itsmeichigo) [inline Yosemite/Yosemite/Stores/AccountCreationStore.swift:45]: ❓ A dumb question, but the `@MainActor` here will guarantee that the `completion` will be triggered on the main thread, right? Will it also cause the network calls to be called on the main thread?

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit: 

Just curious: I assume the account creation requests don't require any authentication - will we need to create an unauthenticated version of `AlamofireNetwork` for this?

File: Yosemite/Yosemite/Stores/AccountCreationStore.swift (hunk 1 above)
Comment (jaclync) [reply Yosemite/Yosemite/Stores/AccountCreationStore.swift:45]: I believe Alamofire executes the actual network request in a background thread, and the [response handling callback in the main thread by default](https://github.com/Alamofire/Alamofire/blob/master/Documentation/Usage.md#response-handler-queue). Other Yosemite stores also make calls to their Remote on the main thread (and actions have to be dispatched on the main thread).

**Comment by jaclync** [comment]
> Just curious: I assume the account creation requests don't require any authentication - will we need to create an unauthenticated version of AlamofireNetwork for this?

Great question! In the next PR, I'm passing an empty auth token to the network like `AlamofireNetwork(credentials: .init(authToken: ""))` but the `AlamofireNetwork` can be improved by allowing multiple ways to authenticate the request. I might leave the improvement for tech debt, in order to meet the timeline.

**Pull request merged**